src/Microsoft.DotNet.Interactive.FSharp/FSharpKernel.fs

```diff
@@ -138,6 +140,122 @@ type FSharpKernelBase () as this =
+                        FsAutoComplete.TipFormatter.FormatCommentStyle.Legacy 
+                    |> Seq.concat
+                    |> Seq.map (fun (signature, comment, footer) ->     
+                        // make footer look like in Ionide
+                        let newFooter = 
+                            footer.Split([|'\n'|], StringSplitOptions.RemoveEmptyEntries) 
+                            |> Seq.filter (fsiAssemblyRx.IsMatch >> not)
+                            |> Seq.map (sprintf "*%s*") |> String.concat "\n\n----\n"
+
+                        let markdown = 
+                            String.concat "\n\n----\n" [
+                                if not (String.IsNullOrWhiteSpace signature) then
+                                    let code =
+                                        match value with
+                                        // don't show function-values
+                                        | Some (Some value) when not (Reflection.FSharpType.IsFunction value.ReflectionType) -> 
+                                            let valueString = sprintf "%0A" value.ReflectionValue
+                                            let lines = valueString.Split([|'\n'|], StringSplitOptions.RemoveEmptyEntries) |> Array.toList
+                                            
+                                            match lines with
+                                            | [] -> 
+                                                signature
+                                            | [line] ->
+                                                sprintf "%s // %s" signature line
+                                            | first :: rest ->
+                                                String.concat "\n" [
+                                                    let prefix = sprintf "%s " signature
+                                                    yield sprintf "%s// %s" prefix first
+
+                                                    let ws = String(' ', prefix.Length)
+
+                                                    for line in rest do
+                                                        yield sprintf "%s// %s" ws line
+                                                ]
+                                        | Some None ->
+                                            sprintf "%s // null" signature
+                                        | _ -> 
+                                            signature
+
+                                    sprintf "```fsharp\n%s\n```" code
+
+                                if not (String.IsNullOrWhiteSpace comment) then
+                                    comment
+
+                                if not (String.IsNullOrWhiteSpace newFooter) then
+                                    newFooter
+                            ]
+
+                        FormattedValue("text/markdown", stdinRx.Replace(fsiModuleRx.Replace(markdown, ""), ""))
```

Comment:
Why are we producing text/markdown?  Does HoverTextProduced require it?

Reply:
It's basically what ionide does and the tooltip-text contains markdown for e.g. generic arguments. Furthermore the F# syntax highlighting works with markdown code-blocks...

Reply:
> Furthermore the F# syntax highlighting works with markdown code-blocks...

Right, though that's a non-standard interpretation of F# code comments only really supported by F# tooling based off the ionide toolchain.

We're talking about standardizing it but it's a long way off, I don't think we should adopt it in F# notebooks